### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by Neotamandua**
- Fix private doc link in rusk-profile
- Fix doc comments in consensus
- Fix outdated doc comment for node-data
- Fix unresolved doc links for execution-core
- Add cargo deny broken doc links to execution-core

### CODE REVIEW

**Review by ureeves — DISMISSED** [review]
I'm honestly shocked there are so few instances of broken links. Thanks for this. Did I see an issue suggesting adding docs to the Makefile somewhere? Perhaps it would be better to simply fail compilation if this happens instead, with:

`#![deny(rustdoc::broken_intra_doc_links)]`

**Comment by Neotamandua** [comment]
> Perhaps it would be better to simply fail compilation if this happens instead

Good idea & I looked at it at the same time you wrote the answer. Just added it

File: consensus/src/merkle.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@ impl<const H: usize> BinaryMerkle<H> {
 /// Calculate the root of a dynamic merkle tree (containing up to 2^15 elements)
 /// in the same way of how dusk-blockchain does
 ///
-/// For reference impl check here https://github.com/dusk-network/dusk-crypto/blob/master/merkletree/merkletree.go
+/// For reference impl check here <https://github.com/dusk-network/dusk-crypto/blob/master/merkletree/merkletree.go>
```
Comment (HDauven) [inline consensus/src/merkle.rs:81]: For completeness, on line 110 in this file there's also a link in the tests that could use the angle brackets.

I'm wondering though if it still makes sense to refer to the Go implementation, but that's probably out of scope.

**Review by HDauven — APPROVED** [review]
LGTM, small comment but you can ignore it if you want.

**Review by ureeves — APPROVED** [review]

**Pull request merged**